src/cascadia/WpfTerminalControl/TerminalContainer.cs

```diff
@@ -0,0 +1,281 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Windows;
+using System.Windows.Interop;
+
+namespace Microsoft.Terminal.Wpf
+{
+    /// <summary>
+    /// Follow steps 1a or 1b and then 2 to use this custom control in a XAML file.
+    ///
+    /// Step 1a) Using this custom control in a XAML file that exists in the current project.
+    /// Add this XmlNamespace attribute to the root element of the markup file where it is 
+    /// to be used:
+    ///
+    ///     xmlns:MyNamespace="clr-namespace:WpfTerminalControl"
+    ///
+    ///
+    /// Step 1b) Using this custom control in a XAML file that exists in a different project.
+    /// Add this XmlNamespace attribute to the root element of the markup file where it is 
+    /// to be used:
+    ///
+    ///     xmlns:MyNamespace="clr-namespace:WpfTerminalControl;assembly=WpfTerminalControl"
+    ///
+    /// You will also need to add a project reference from the project where the XAML file lives
+    /// to this project and Rebuild to avoid compilation errors:
+    ///
+    ///     Right click on the target project in the Solution Explorer and
+    ///     "Add Reference"->"Projects"->[Browse to and select this project]
+    ///
+    ///
+    /// Step 2)
+    /// Go ahead and use your control in the XAML file.
+    ///
+    ///     <MyNamespace:TerminalControl/>
+    ///
+    /// </summary>
+    public class TerminalContainer : HwndHost
+    {
+        private ITerminalConnection connection;
+        private IntPtr hwnd;
+        private IntPtr terminal;
+
+        private NativeMethods.ScrollCallback scrollCallback;
+        private NativeMethods.WriteCallback writeCallback;
+
+        // We keep track of the DPI scale since we could get a DPI changed event before we are able to initialize the terminal.
+        private DpiScale dpiScale = new DpiScale(96, 96);

```

Comment:
Can we put 96 as the USER_DEFAULT_SCREEN_DPI constant somewhere and use it twice here?